### PR TITLE
Add a spec for the sandbox images for runtime analysis.

### PR DESCRIPTION
**Pull request by calebbrown**
WIP

This doc defines how Sandbox container images are expected to behave.

### CODE REVIEW

File: sandboxes/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+#### Arguments
+
+- Must take 2 required arguments and 1 optional argument:
+    1. phase name
+        - must support at least "all", "install".
+        - may also include "import" and any other phases useful to the
+          ecosystem.
+    1. package:
+        - either a local file contain a package to install;
+        - or the name of the package in the package repository to install.
```
Comment (oliverchang) [inline sandboxes/README.md:24]: Should we add an explicit '--local' flag to make distinguishing between these two cases more explicit?

File: sandboxes/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+- the analysis command
+- the package manager used by the analysis command
+- system libraries used by the majority of packages
+
+### Wiring It Up
+
+1. Update [build_docker.sh](../build/build_docker.sh) to reference the image.
+2. Extend [internal/pkgecosystem](../internal/pkgecosystem) to add support for
+   the new sandbox.
+3. Ensure your new ecosystem is supporteded by
```
Comment (oliverchang) [inline sandboxes/README.md:121]: nit: spelling of `supported`

**Review by oliverchang — APPROVED** [review]

File: sandboxes/README.md (hunk 1 above)
Comment (calebbrown) [reply sandboxes/README.md]: Added. I was iffy about this, but I think being explicit is better than implicit.

File: sandboxes/README.md (hunk 2 above)
Comment (calebbrown) [reply sandboxes/README.md]: Done

**Pull request merged**